test/unittests/execute_test.cpp

```diff
@@ -664,114 +454,99 @@ TEST(execute, i64_store_overflow)
-    module.codesec.emplace_back(
-        Code{0, {Instr::local_get, Instr::local_get, Instr::i32_store8, Instr::end},
-            {1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}});
-
-    auto instance = instantiate(module);
-    const auto [trap, ret] = execute(instance, 0, {0xf1f2f380, 65537});
-
-    ASSERT_TRUE(trap);
-}
-
-TEST(execute, i32_store16)
-{
-    Module module;
-    module.memorysec.emplace_back(Memory{{1, 1}});
-    module.funcsec.emplace_back(TypeIdx{0});
-    module.codesec.emplace_back(
-        Code{0, {Instr::local_get, Instr::local_get, Instr::i32_store16, Instr::end},
-            {1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}});
-
-    auto instance = instantiate(module);
-    const auto [trap, ret] = execute(instance, 0, {0xf1f28000, 0});
-
-    ASSERT_FALSE(trap);
-    ASSERT_EQ(ret.size(), 0);
-    ASSERT_EQ(instance.memory->substr(0, 4), from_hex("00800000"));
-
-    ASSERT_TRUE(execute(instance, 0, {0xf1f28000, 65537}).trapped);
-}
-
-TEST(execute, i64_store8)
-{
-    Module module;
-    module.memorysec.emplace_back(Memory{{1, 1}});
-    module.funcsec.emplace_back(TypeIdx{0});
-    module.codesec.emplace_back(
-        Code{0, {Instr::local_get, Instr::local_get, Instr::i64_store8, Instr::end},
-            {1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}});
-
-    auto instance = instantiate(module);
-    const auto [trap, ret] = execute(instance, 0, {0xf1f2f4f5f6f7f880, 0});
-
-    ASSERT_FALSE(trap);
-    ASSERT_EQ(ret.size(), 0);
-    ASSERT_EQ(instance.memory->substr(0, 8), from_hex("8000000000000000"));
+    /* wat2wasm
+    (memory 1 1)
+    (func (param i32 i32)
+      get_local 1
+      get_local 0
+      i32.store  ;; to be replaced by variants of i32.store
+    )
+    */
+    const auto wasm =
+        from_hex("0061736d0100000001060160027f7f00030201000504010101010a0b010900200120003602000b");
+    auto module = parse(wasm);
 
-    ASSERT_TRUE(execute(instance, 0, {0xf1f2f4f5f6f7f880, 65537}).trapped);
-}
+    auto& store_instr = module.codesec[0].instructions[2];
+    ASSERT_EQ(store_instr, Instr::i32_store);
+    ASSERT_EQ(module.codesec[0].immediates.substr(8), "00000000"_bytes);
 
-TEST(execute, i64_store16)
-{
-    Module module;
-    module.memorysec.emplace_back(Memory{{1, 1}});
-    module.funcsec.emplace_back(TypeIdx{0});
-    module.codesec.emplace_back(
-        Code{0, {Instr::local_get, Instr::local_get, Instr::i64_store16, Instr::end},
-            {1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}});
+    constexpr std::tuple<Instr, unsigned> test_cases[]{
+        {Instr::i32_store8, 8},
+        {Instr::i32_store16, 16},
+        {Instr::i32_store, 32},
+    };
 
-    auto instance = instantiate(module);
-    const auto [trap, ret] = execute(instance, 0, {0xf1f2f4f5f6f78000, 0});
+    for (const auto& test_case : test_cases)
+    {
+        const auto num_bits = std::get<1>(test_case);
+        constexpr auto arg_background = uint32_t{0xf1f2f3f4};
+        const auto arg_value = (uint32_t{1} << (num_bits - 1)) | 1u;  // 0x80...01
```

Comment:
These calculations look complicated, maybe it would be simpler just to hard-code `arg`  and `expected_memory` in `test_cases`?